### PR TITLE
Simplify result object

### PR DESCRIPTION
**Pull request by omarkmu**
### Features
- Added the `explicitArray: false` option to the parser.
- Added type conversions to account for expected array types and numbers.
- Added the `trim: true` option to the parser.
  - Removed `handleTabAndNewline`; this is now handled by the trim option.
- Modified the result structure.
  - `requestParameters` is now added directly to the result.
  - `channel` is renamed to `data` to match the existing key.
- Updated existing tests to reflect result simplification.
- Added some simple tests for dictionaryView (perhaps not enough).

### Fixes
- Fixed multiple select bug for various types:
  - TranslationLanguage
  - FirstCategory
  - SecondCategory
  - VocabularyGrade
  - PartOfSpeech
  - MultimediaInfo
  - MeaningCategory *(edit: as it turns out, the API only accepts one value for this type)*
  - SubjectCategory
- Changed `homomorphicNum` to `homographNum` for clarity.
  - I think that name came about from a mistranslation on my part at some point—sorry!

### Suggestions
1. Changing the aliases of `firstCategory` (type1) and `secondCategory` (type2). These names aren't particularly descriptive (although better than "type1" and "type2"), and I figure a major version increase is the perfect time to change them. Instead of `firstCategory`, `classification` may be a better fit (this is the word used on the website's advanced search for the same option). `secondCategory` is technically "classification 2", but `originType` may be a more descriptive name for it.
2. Changing the aliases of `searchTarget` (part) and `searchTargetType` (target). The documentation isn't very helpful for figuring out the difference between these, considering the descriptions of both translate loosely to "search target", but checking the query parameters in the website's URL shows that 'part' corresponds to `searchType` in the query parameters. 'target', on the other hand, corresponds to the search target in the advanced search; `searchTarget` would be a better name for *this* parameter.

TL;DR (I completely understand if you skipped to this, that was a wordy explanation), I'd like to rename these parameters:
- `firstCategory` → `classification`
- `secondCategory` → `originType`
- `searchTarget` → `searchType`
- `searchTargetType` → `searchTarget`

I'll go ahead and add a commit fixing the array problem and hold off on the renaming until I hear back about it.
@Fox-Islam

### CODE REVIEW

**Comment by omarkmu** [comment]
I'm unsure if this is a permanent change, as the documentation has not yet been updated, but I thought I should notify you nonetheless: The 'method' query parameter of the /view endpoint seems to have been changed to 'view_method' during this weekend's maintenance, which breaks target code view queries. Probably best to wait and see if the change is reverted before making any changes or merging this. @Fox-Islam 

Edit: The admin responded to my bug report; the change was unintentional and has been reverted.

**Comment by Fox-Islam** [comment]
Aaaaah, sorry for leaving this for so long, at some point I must have cleared the notifications and never thought to check back until now. This is great work! I'd like to take a moment to try it out and see how things look in case any changes have gone into the API since October, hopefully I'll get a chance in the morning! 

As for the suggested alias changes, they all sound very reasonable, would definitely make much more sense than they do now 😅

**Comment by omarkmu** [comment]
@Fox-Islam No worries, it happens! I'll get around to making those changes when I get the chance.

**Comment by omarkmu** [comment]
Hey @Fox-Islam, I just ran the tests and this package seems to have the same problem I'm currently having with [krdict.py](https://github.com/omarkmu/krdict.py). I'm not sure why, but the Python Requests library (and apparently axios, too) are now unable to verify the certificates.

The website still shows the "connection secure" lock indicator when visited, and the same is true for API result pages, so I don't think the issue is on their end. I haven't had much time to look into this yet, but hopefully it can be solved without too much of a headache.

**Update:**
Not too much of a headache, but it's a less than ideal solution. I couldn't figure out any way to solve it that doesn't involve including a PEM file (well, other than turning off SSL verification, but I don't think that would be very responsible). I tried the intermediate certificate, "Thawte RSA CA 2018", but that didn't work, which is a shame because that one doesn't expire until 2027, whereas the one that *works* expires at the end of this year (based on its validity start range, it only lasts about a year).

So, it's a solution, but hopefully only a temporary one. I don't know what the corresponding option is in axios, but in Python I only had to update the request calls to include the path of the PEM file. I've included the contents of the file [here](https://gist.github.com/omarkmu/b498d2ad0c949609619e457214119619).
